common/modules/token_send/src/token_send.rs

```diff
@@ -54,11 +96,126 @@ pub trait TokenSendModule {
             nonce,
             amount,
             gas_limit,
-            &function,
+            &ManagedBuffer::managed_from(self.type_manager(), function),
+            &ManagedArgBuffer::new_empty(self.type_manager()),
+        ))
+    }
+
+    fn send_multiple_tokens(
+        &self,
+        destination: &Address,
+        payments: &[EsdtTokenPayment<Self::Api>],
+        opt_accept_funds_func: &OptionalArg<BoxedBytes>,
+    ) -> SCResult<()> {
+        let (function, gas_limit) = match opt_accept_funds_func {
+            OptionalArg::Some(accept_funds_func) => (
+                accept_funds_func.as_slice(),
+                self.transfer_exec_gas_limit().get(),
+            ),
+            OptionalArg::None => {
+                let no_func: &[u8] = &[];
+                (no_func, 0u64)
+            }
+        };
+
+        SCResult::from_result(self.raw_vm_api().direct_multi_esdt_transfer_execute(
```

Comment:
can't you use this instead of defining the C API ?

Reply:
it will not work due to wasm compile error